File: addons/guide-cpp/src/inputs/guide_input_touch_axis_2d.cpp
#include "guide_input_touch_axis_2d.h"

using namespace godot;

void GUIDEInputTouchAxis2D::_bind_methods() {
    ClassDB::bind_method(D_METHOD("_to_string"), &GUIDEInputTouchAxis2D::_to_string);
}

void GUIDEInputTouchAxis2D::_apply_value(Vector2 value) {
    _value = Vector3(value.x, value.y, 0);
}

bool GUIDEInputTouchAxis2D::is_same_as(const Ref<GUIDEInput> &other) const {
    Ref<GUIDEInputTouchAxis2D> o = other;
    if (o.is_null()) return false;
    return o->get_finger_count() == finger_count && 
           o->get_finger_index() == finger_index;
}

String GUIDEInputTouchAxis2D::_to_string() const {
    return "(GUIDEInputTouchAxis2D finger_count=" + String::num(finger_count) + 
           " finger_index=" + String::num(finger_index) + ")";
}

String GUIDEInputTouchAxis2D::_editor_name() const {
    return "Touch Axis2D";
}

String GUIDEInputTouchAxis2D::_editor_description() const {
    return "2D relative movement of a touching finger.";
}

GUIDEAction::GUIDEActionValueType GUIDEInputTouchAxis2D::_native_value_type() const {
    return GUIDEAction::AXIS_2D;
}

File: addons/GuideCS/Inputs/GUIDEInputTouchAxis2D.cs
using System;
using Godot;

namespace Guide;

[Tool]
[GlobalClass]
public partial class GUIDEInputTouchAxis2D : GUIDEInputTouchAxisBase {
	protected override void _ApplyValue(Vector2 value) {
		_Value = new Vector3(value.X, value.Y, 0);
	}

	public override bool IsSameAs(GUIDEInput other) {
		return other is GUIDEInputTouchAxis2D && base.IsSameAs(other);
	}

	public override string ToString() => $"(GUIDEInputTouchAxis2D finger_count={FingerCount} finger_index={FingerIndex})";

	public override string _EditorName() => "Touch Axis2D";

	public override string _EditorDescription() => "2D relative movement of a touching finger.";

	public override GUIDEAction.GUIDEActionValueType _NativeValueType() => GUIDEAction.GUIDEActionValueType.Axis2D;
}
